### ISSUE TITLE
`yarn install` fails inside /mnt with EINVAL for lstat during step 3

### ISSUE THREAD

**Issue by khuongduybui** [Microsoft/BashOnWindows#2448]
* Your Windows build number:  Version 10.0.16273.1000

* What you're doing and what's happening:  (Copy&paste specific commands and their output, or include screen shots)

Running yarn (install) in a directory inside /mnt/c/Users/foo/bar
```
$ pwd
/mnt/c/Users/duybui/code/web/old
$ yarn
yarn install v0.27.5
warning package.json: No license field
info No lockfile found.
warning aurelia-web@1.0.0: No license field
[1/4] Resolving packages...
warning aurelia-cli > npm > request > node-uuid@1.4.8: Use uuid module instead
warning browser-sync > browser-sync-ui > weinre > express@2.5.11: express 2.x series is deprecated
warning browser-sync > browser-sync-ui > weinre > express > connect@1.9.2: connect 1.x series is deprecated
[2/4] Fetching packages...
warning fsevents@1.1.2: The platform "linux" is incompatible with this module.
info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "muicss@0.9.22" has unmet peer dependency "react@^0.14.0 || ^15.0.0".
error An unexpected error occurred: "EINVAL: invalid argument, lstat '/mnt/c/Users/duybui/code/web/old/node_modules/accord/node_modules/cliui/README.md'".
info If you think this is a bug, please open a bug report with the information provided in "/mnt/c/Users/duybui/code/web/old/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

* What's wrong / what should be happening instead:
Note that every time I run the command it may fail while lstat'ing a different file.

**Comment by khuongduybui** [comment]
My current workaround is to create a node_modules directory outside /mnt, then `ln -s` it into my directory inside /mnt.

**Comment by suitupalex** [comment]
Related issue in the Yarn repo: yarnpkg/yarn#4188

**Comment by hikurangi** [comment]
Also getting this. I can install packages successfully using npm, so am I correct in thinking that this is a Yarn issue, not a WSL?

**Comment by khuongduybui** [comment]
It's a WSL issue with the recent file system later changes

**Comment by hikurangi** [comment]
[I see](https://github.com/yarnpkg/yarn/issues/4188).

**Comment by tjmonsi** [comment]
It fails on the new build of windows. I reverted back to 16251.rs3 and yarn works fine again

**Comment by sunilmut** [comment]
Adding @SvenGroot who is taking a look at this. Also, marking it as a bug.

**Comment by velagand** [comment]
Also fails on 16278.rs3, npm works fine

**Comment by simonbuchan** [comment]
This looks like it's the same error in React Native's packager (roughly equivalent to webpack --watch):

```
✘130 ➜ yarn start
yarn start v1.0.1
$ node node_modules/react-native/local-cli/cli.js start
Scanning 723 folders for symlinks in /mnt/c/code/learning-hub/mobile-app/node_modules (80ms)
 ┌────────────────────────────────────────────────────────────────────────────┐
 │  Running packager on port 8081.                                            │
 │                                                                            │
 │  Keep this packager running while developing on any JS projects. Feel      │
 │  free to close this tab and run your own packager instance if you          │
 │  prefer.                                                                   │
 │                                                                            │
 │  https://github.com/facebook/react-native                                  │
 │                                                                            │
 └────────────────────────────────────────────────────────────────────────────┘
Looking for JS files in
   /mnt/c/code/learning-hub/mobile-app

Loading dependency graph...
React packager ready.

Loading dependency graph, done.
Bundling `index.android.js`
  Transforming modules  100.0% (913/913), done.
 ERROR  EINVAL: invalid argument, lstat '/mnt/c/code/learning-hub/mobile-app/.git/index.lock'
{"errno":-22,"code":"EINVAL","syscall":"lstat","path":"/mnt/c/code/learning-hub/mobile-app/.git/index.lock"}
Error: EINVAL: invalid argument, lstat '/mnt/c/code/learning-hub/mobile-app/.git/index.lock'
    at Error (native)

See http://facebook.github.io/react-native/docs/troubleshooting.html
for common problems and solutions.
error Command failed with exit code 11.
```

This one is pretty intermittent though.

**Comment by sunilmut** [comment]
We have been trying to get to the bottom of this issue, but the bug seems very evasive. Every time we try to gather an strace or put it under the debugger, it does not repro. Seems like some kind of race condition. If someone is able to get a strace of the failing command, that might shed some light. `strace -o <trace_file> -ff <failing_cmd>`

**Comment by AndyGauge** [comment]
I attempted to create a react app using yarn `create-react-app hello-api` and ran it through the strace recommended.  I zipped up the output.  

https://github.com/AndyGauge/yarn-debug/blob/master/yarn-fail.zip

**Comment by khuongduybui** [comment]
When I run `strace -o ~/trace.log yarn install` the error does NOT show up.
When I run `strace -o /dev/null yarn install` the error DOES show up.

**Comment by AndyGauge** [comment]
`rails webpack:install` fails with the same message.  The trace.log file: https://github.com/AndyGauge/yarn-debug/blob/master/trace.log

**Comment by CrshOverride** [comment]
I was able to reproduce locally. You can download a zip of all of the trace files at https://www.dropbox.com/s/oy78n0qvtqqyxwb/yarn_trace.zip?dl=0

**Comment by hikurangi** [comment]
If it helps, for me the problem seems to be resolved in 16288.1.

**Comment by khuongduybui** [comment]
No the *original* issue has absolutely NOT been resolved in latest insider (non skip-ahead) build.

**Comment by wycats** [comment]
I was able to reproduce this issue with strace on.

[trace.zip](https://github.com/Microsoft/BashOnWindows/files/1315497/trace.zip)

**Comment by SvenGroot** [comment]
Thanks for all the input. We've been able to determine the root cause of this issue and are working on a fix.

**Comment by khuongduybui** [comment]
You made my (our) day! Keep up the good work.

**Comment by Warblefly** [comment]
The news of the cause being determined is very, very welcome. I shall be glad to test it. Maybe the fix is in build 16296, which just landed here?

**Comment by AndyGauge** [comment]
I was upgraded to 16296 yesterday and still had to work around the issue by moving across file systems to run `yarn add` I'm sure it still affects `yarn install`.

**Comment by SvenGroot** [comment]
We're still working on the fix, so it's not available yet. I'll update this thread once the fix becomes available.

**Comment by jingsam** [comment]
This issue has annoyed me for a month, expecting fix it in the next release.

**Comment by Verikon** [comment]
@jingsam agree, this one is particularly painful.

**Comment by orenfromberg** [comment]
I am having this error on Windows 10 Home 1709 16299.15

**Comment by khuongduybui** [comment]
I upgraded to 17017.1000 and no longer observe this issue. Can someone confirm a fix?

**Comment by Archomeda** [comment]
Updated my Windows 10 Pro yesterday to 16299.15, which was already painful with .url shortcuts not working, night light having weird coloration, Nvidia driver 387.68 shitting themselves (why even install this version automatically, it's not even available from Nvidia themselves), and now WSL not handling `yarn add` properly.

This really is the worst Windows update for me so far. I'm reverting back to the previous Windows 10 version. Any ETA for when this is fixed?

**Comment by orenfromberg** [comment]
@khuongduybui I also upgraded to 17017.1000 and the issue is no longer observed. Thanks!

**Comment by SkyZH** [comment]
I have upgraded to 17017.1000 but this issue still exists.

```skyzh@SkyZH-RESCUER-PC:/mnt/d/Work/BlueSense$ yarn install
yarn install v0.27.5
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.1.2: The platform "linux" is incompatible with this module.
info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "bootstrap@4.0.0-beta" has unmet peer dependency "jquery@>=3.0.0".
warning "bootstrap@4.0.0-beta" has unmet peer dependency "popper.js@^1.11.0".
error An unexpected error occurred: "EINVAL: invalid argument, lstat '/mnt/d/Work/BlueSense/node_modules/@firebase/util/gulpfile.js'".
info If you think this is a bug, please open a bug report with the information provided in "/mnt/d/Work/BlueSense/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by khuongduybui** [comment]
@skyzh I am using latest version of yarn 1.x and do not observe the problem. Maybe that could help.

**Comment by Verikon** [comment]
Running 1709 (OS Build 17017.1000) and still seeing the issue;
```
bren@Crowley:/mnt/e/project$ yarn install
yarn install v1.1.0
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.1.2: The platform "linux" is incompatible with this module.
info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
error An unexpected error occurred: "EINVAL: invalid argument, lstat '/mnt/e/project/node_modules/amqplib/node_modules/string_decoder/index.js'".
info If you think this is a bug, please open a bug report with the information provided in "/mnt/e/project/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
bren@Crowley:/mnt/e/project$
```

**Comment by SvenGroot** [comment]
Build 17017 does not have the fix for this issue yet. The fix has been checked in and is making its way through our system, and will hopefully be in an insider build sooner rather than later, but it's difficult for me to say when that will happen. We'll keep you informed.

**Comment by codeart1st** [comment]
Is it possible that fixes like that will be patched with normal windows updates?

**Comment by danicatalan** [comment]
This is working for me now after updating to 17017.rs_prerelease.171010-1400 with `yarn 1.0.2`.

**Comment by mrmckeb** [comment]
@SvenGroot I installed the official release today and encountered this problem. I understand this will go out to insiders, but what about those of us not in the insider program - will this be available as an update through Windows Update or the Microsoft Store?

We rely on Yarn Workspaces - so this effectively stopped me from being able to work on our project, unless I roll back Windows 10 (which I might do tonight).

**Comment by nelak** [comment]
@SvenGroot  I was just hit by this when upgrading to the RTM fall update.
Can you confirm if there's an ETA for the fix and if it will be possible to get it without going through the insiders builds?

**Comment by SvenGroot** [comment]
We are working to get this bug serviced in the Fall Creator's Update as soon as possible. Unfortunately, the timing of that depends on a lot of things besides just me, so I can't give you any dates. We'll keep you posted.

**Comment by nelak** [comment]
Thanks for the update, and looking forward to the fix.

**Comment by Warblefly** [comment]
Still getting failures in compilation of very large projects within lxss with gcc very occasionally reporting EINVAL. Re-running the compilation cures the problem, so it appears that the referenced file (a header, usually) is still occasionally turning into something odd, then coming back again. Built 17017.1000. This is a terrible pity, because WSL used to be totally reliable for the biggest compilations, like native GNU/Linux. Very sad to see it not working as it used to.

**Comment by SvenGroot** [comment]
Unfortunately, Insider build 17025 does not appear to have the fix for this yet. However, it looks like it will be in the next Insider preview build. We're still working on getting this serviced on the Fall Creator's Update as well.

**Comment by Warblefly** [comment]
Thanks for letting me know.

**Comment by Log1x** [comment]
😿

**Comment by noinkling** [comment]
For some of us this issue is a massive showstopper and causing more than a little frustration. Don't forget that it's also a regression (at least from a user perspective), so I would expect it to be very high-priority. It has been about 3 weeks since the FCU landed and we haven't gotten much more than a vague "we're working on it", which doesn't exactly inspire a whole lot of confidence that this will be addressed in a timely manner. We would greatly appreciate a little more information/assurance about when/how the fix is going to land, so we can at least make better decision on how to mitigate the productivity loss caused in the meantime. Any insight into the process, even if it's just a hypothetical worst-case scenario, is better than having a question mark hanging over the whole thing (if it's not going to be speedy).

I don't want to sound ungrateful or anything - I'm very appreciative of the work that has gone into WSL, and from what I understand things like update timing are mostly outside the WSL team's control - but a little more transparency would go a long way to alleviate some of our concern.

**Comment by rhyek** [comment]
This sort of thing paints Windows + WSL as an unreliable setup for modern Web development where Linux or MacOS seem more robust and streamlined. I understand Microsoft's interest in making something like WSL, especially with new hardware offerings such as the Surface Book 2 competing directly with Macbook Pros. They want to target professional users who use their laptops as their main work tool, but there doesn't seem to be enough focus going into making WSL attractive/trustworthy enough with issues such as this persisting almost 2.5 months after being reported.

How can I, as a professional, justify purchasing Microsoft products when I can't trust them to properly support them in a timely manner?

Sorry for the rant. I really want my next laptop to be a Surface Book 2, but it is not looking very likely.

**Comment by peterjuras** [comment]
The rants might seem harsh, but maybe to reiterate the impact of this issue:

Right now it is not possible to work on **any** nodejs, webpack, etc. project since both npm and yarn fail to install. The workaround is very painful and costs too much time because it needs to be done on each dependency change or adjustment.

I'm actually quite pessimistic, there were no WSL updates between Creator's Update and Fall Creator's Update for users of the stable Windows version as far as I can remember, therefore it is somewhat sad to have an absolutely broken WSL for node users, especially with the big fanfare of blog posts announcing that WSL would no longer be beta?

**Comment by Log1x** [comment]
The above couldn't explain the situation any better. I had been tempted to write something similar but wasn't sure I could contain my anger and frustration in the process.

The amount of time (AND MONEY) I've lost the last 3 weeks trying to work around this bug has made me incredibly bitter after spending hours writing guides, resources, tutorials, etc. talking up WSL.

![Screenshot](https://i.imgur.com/cbAxSIQ.png)

I have the #1 result on Google for a majority of the keywords revolving around setting up development with Windows and it makes me incredibly tempted to just edit my post and tell people to stay the hell away.

If Microsoft can't take this seriously, I'm afraid I can't take them seriously either.

**Comment by orenfromberg** [comment]
I had been developing with WSL since January 2017. This issue forced me to install and dual boot with Ubuntu. I managed to fix the issue on my personal laptop but the issue persisted on my main work laptop. It will be a hard sell for me to go back to developing on Windows after this issue.

**Comment by simonbuchan** [comment]
I should note that for many npm workflows, including the majority of webpack setups, you can simply use Windows yarn or npm, and that it's possible to alias them such that you don't even need to change your workflow.

That said https://blogs.msdn.microsoft.com/commandline/2017/07/28/windows-subsystem-for-linux-out-of-beta/ is pretty ironic 😓

**Comment by rhyek** [comment]
@simonbuchan 

> I should note that for many npm workflows, including the majority of webpack setups, you can simply use Windows yarn or npm, and that it's possible to alias them such that you don't even need to change your workflow.

Working with 2 filesystems at the same time is already hacky enough to be working with two redundant binaries for each OS simultaneously as well. Which to use when and how? I would rather just have everything inside the Linux context and have it work 100%. That should be the goal.

**Comment by noinkling** [comment]
@simonbuchan That may work ok for some people, but be warned that it doesn't play nice with native modules, binary/binstub formats, and packages that make use of inotify (or anything else that requires OS checks at install time).

The other workaround to have node_modules reside on the WSL filesystem is probably better, but that obviously breaks any Windows tooling (e.g. VS Code intellisense).

**Comment by Verikon** [comment]
I must say, my Vim skills have greatly increased due to this issue exclusively.

Being this issue has arisen from Yarn , its probably safe to assume the people most pained from this issue are ReactJS developers (and some Angular). This issue has really empowered the "nay-windows sayers" in my workplace when they argue "you cannot develop javascript on windows, in the same way you cannot really develop .NET applications without windows".

I'm a big advocate from win10 and personally feel the new direction and innovations from Microsoft should be both acknowledged and enjoyed because really, its just a great OS!

Please take this as a strong use case to improve the release methodology for WSL or we'll loose the opportunity to enjoy it as an option in our environment.

**Comment by jazoom** [comment]
Apparently my Surface Pro 4 has updated itself overnight because now I have the same issue. It's sad that this issue was surfaced 2.5 months ago and it still is hitting new people.

**Comment by bitcrazed** [comment]
Hi all. I've been pinged several times over the last 48 hours for an update on the fix for this issue.

**Know that we appreciate and understand the enthusiasm for the fix for this issue, but we do ask for your patience as the fix makes its way up through our build and test systems, on its way to the Insider rings.** 

As @SvenGroot stated above, the fix has been checked in but it didn't make it up to the Insider release branch in time for [17025 which was released on 11/1](https://twitter.com/donasarkar/status/925769980150345732). It's v. likely it'll arrive in the next Insider build.

Sven is also [working to backport this fix to Fall Creators Update (FCU)](https://github.com/Microsoft/WSL/issues/2448#issuecomment-339435525), but since this isn't a fix for a major security exploit, it'll make its way through our normal servicing channels, passing through numerous review and testing gates before it's formally released in an up-coming servicing release.

## Please note...
While Microsoft is much more agile today than ever before (we now essentially ship Windows every week) there are still processes that churn behind the scenes to ensure that we catch as many bugs as we can before they're released, and that fixes are tested & released in an organized manner. 

Only fixes for the most high-risk, high-impact issues bypass these standard processes, but these exceptional fixes are themselves placed under the most exacting scrutiny possible, requiring HUGE deployment of humanpower & tooling to quadruple-check that the fix is good before its released. We reserve these rapid fix deployments only for the most important fixes for the most critical of issues, and while annoying, this fix does not qualify as 'exceptional'.

No software is ever perfect or bug free, and when you're delivering software used by more than a billion people every day, one has to take whatever steps possible to ensure the best outcome for the largest audience. The last thing we'd want is to release a fix that doesn't resolve the issue, and that may even make things worse.

So, please bear with us while we process this fix through our release processes. 

Thanks.

**Comment by SvenGroot** [comment]
We appreciate the pain some of you are feeling because of this issue. Please believe me: we take this very seriously and have been working very hard to get the fix to you as soon as possible.

In the mean time, the following workaround should get around this issue. When starting bash, run the following commands:
`sudo umount /mnt/x`
`sudo mount -t drvfs -o noatime,fallback=1 X: /mnt/x`

Replace /mnt/x and X: with the mount point and drive letter for any affected drives. If you've done this correctly, you can run `mount` and you'll see a line like this:
`C: on /mnt/c type drvfs (rw,noatime,fallback=1)`

If the fallback option is shown, it has been mounted correctly. We believe that doing this will allow you to use the drvfs mount without hitting this issue.

You'll have to do this every time you start a new WSL instance (so every time you open bash after all bash windows were closed).

**Note! Doing this will disable all the performance improvements we've implemented since RS1 (version 1607, the Anniversary Update). Don't use this workaround unless you're hitting this issue.**

I'll update once the fix is released, at which point I strongly recommend you stop using this workaround.

**Comment by Melatonin64** [comment]
@SvenGroot you are our savior, thank you!
I've added the workaround to my `~/.bashrc` and it seems to work:

```bash
# Workaround for issue: https://github.com/Microsoft/WSL/issues/2448
if ! mount | grep -q "C: on /mnt/c type drvfs (rw,noatime,fallback=1)"; then
        echo "== Remount of C: drive required =="
        pushd ~ > /dev/null
        sudo umount /mnt/c
        sudo mount -t drvfs -o noatime,fallback=1 C: /mnt/c
        popd > /dev/null
fi
```

**Comment by JimW** [comment]
Thanks for this fix.  Just to confirm my understanding of when this fix is necessary (for usage with Yarn), it's only necessary when performing certain functionality of Yarn? Like when creating and updating dependencies?

I've wrapped the above code within a prompt so that my WSL is not always crippled back to version 1607.

**Comment by manuelbieh** [comment]
Workaround bei @Melatonin64 works fine as well as the workaround by @khuongduybui (symlink outside /mnt). Quick bugfix would be appreciated though! Thanks for the support here @ wsl team!

**Comment by noinkling** [comment]
Thanks a lot for the workaround, that will definitely tide me over until the fix is available.

@JimW
I think it will depend on what you're specifically doing, but anecdotally/personally I haven't run into the issue (yet) outside `yarn install/upgrade`. The rest of my day-to-day workflow works fine on the new version, so I won't be using the workaround the vast majority of the time. So yeah, it sounds like you have the right idea.

**Comment by imgwho** [comment]
I'm really very sad i meet the same problem today.When i use create-react-app then this problem appear.Hope mircrosoft fix this bug soon.....
sad

**Comment by khuongduybui** [comment]
@imgwho in latest insider release I have not observed the issue, using yarn 1.x, but there are other workarounds, especially https://github.com/Microsoft/WSL/issues/2448#issuecomment-342254181, that could help.

**Comment by JimW** [comment]
@imgwho, create-react-app will successfully complete with the above workaround code, using the latest regular "Outsider" Windows.

**Comment by imgwho** [comment]
@khuongduybui  thx ~~ i'll try

**Comment by imgwho** [comment]
@JimW  oh  i update the create-react-app to 1.4.3 then i found the problem sloved ,, thx~~~

**Comment by hikurangi** [comment]
@imgwho and whoever else it might help, there's yarn's built-in `yarn create react-app myapp`, meaning you no longer need the independent `create-react-app` package...

**Comment by SvenGroot** [comment]
This bug should be fixed in Insider build 17035, which was just released. Please stop using the workaround posted above, and let us know if you still encounter any issues.

**Comment by Log1x** [comment]
Will an update for this be possible to push to non-insider or are those of us not on insider effectively screwed this update?

**Comment by imgwho** [comment]
@hikurangi  i'm Fall Creator's Update (build 16299.19) use WSL on Ubuntu 16.04.3 LTS ,i used use npm to install yarn and then i use apt-get to install yarn
[yarn official website](https://yarnpkg.com/en/docs/install#linux-tab)
```
Debian/Ubuntu Linux

On Debian or Ubuntu Linux, you can install Yarn via our Debian package repository. You will first need to configure the repository:

curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
On Ubuntu 16.04 or below and Debian Stable, you will also need to configure the NodeSource repository to get a new enough version of Node.js.

Then you can simply:

sudo apt-get update && sudo apt-get install yarn
Note: Ubuntu 17.04 comes with cmdtest installed by default. If you’re getting errors from installing yarn, you may want to run sudo apt remove cmdtest first. Refer to this for more information.
```
and then i found..  oops.... 
```
wuv@wuv:/mnt/e/practice/react$ yarn create react-app app
yarn create v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Installed "create-react-app@1.4.3" with binaries:
      - create-react-app
[##############################################################################] 107/107 Creating a new React app in /mnt/e/practice/react/app.

Installing packages. This might take a couple of minutes.
Installing react, react-dom, and react-scripts...

yarn add v1.3.2
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.1.2: The platform "linux" is incompatible with this module.
info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
error An unexpected error occurred: "EINVAL: invalid argument, lstat '/mnt/e/practice/react/app/node_modules/acorn-jsx/index.js'".
info If you think this is a bug, please open a bug report with the information provided in "/mnt/e/practice/react/app/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.

Aborting installation.
  yarnpkg add --exact react react-dom react-scripts --cwd /mnt/e/practice/react/app has failed.

Deleting generated file... node_modules
Deleting generated file... package.json
Deleting generated file... yarn-error.log
Deleting app / from /mnt/e/practice/react
Done.
error Command failed.
Exit code: 1
```
so i think i should just waiting the 17035 come...

**Comment by SvenGroot** [comment]
@Log1x As we've pointed out before, a fix for the Fall Creator's Update will be coming soon. Please be patient, and continue to use the workaround I posted above in the mean time. I will update here once the fix has been released.

**Comment by Stanzilla** [comment]
The fix was in 17035 so all current Insiders builds should no longer have this issue.

**Comment by bitcrazed** [comment]
@orenfromberg @peterjuras @Log1x and anyone else reporting this issue - can you please upgrade to Insiders fast-ring build 17035 (or later) - or install a VM running this build - and let us know if your workflow is now fixed & unblocked?

TIA.

**Comment by rixrix** [comment]
hello,

The last insider build fixed the issue on my machine.  Below is my environment:

```
Yarn: v1.3.2
Node.js: v6.9.5, v8.6.0
Insider Build Preview OS Build:  17035.1000
Insider Build Preview Version:    1709 
```
sneakily ran `npm install` works as expected however it just made the office internet choked up a little bit but all good. 

thanks for all your hard work!

**Comment by abattenburg** [comment]
After updating the issue seems to be fixed for me!

**Comment by bitcrazed** [comment]
Great, thanks for testing and sharing your results. We (i.e. @svengroot ;)) continue to backport the fix through our release machinery.

**Comment by mrmckeb** [comment]
@bitcrazed @SvenGroot Thanks so much for your work on this. As developers, we can all appreciate how tricky these things can be to solve and the effort required to backport a fix.

Is there currently an ETA on this? And if not, an approximation - one week, a month?

**Comment by RinatMullayanov** [comment]
@bitcrazed @SvenGroot Please, tell me, in what **not** insider build will this bug be fixed?

**Comment by Warblefly** [comment]
Using Insider Build 17040 on my very large compilation executed on WSL, with the compilation taking place on a DrvFs mount, almost everything works.

At one point in the day-and-a-half long cross-compiled on eight processors on a 16GB machine, I received:

x86_64-w64-mingw32-gcc: command not found

Once, and once only. Restarting the make process got past this point.

The "x86_64-w64-mingw32-gcc" is my cross compiler, as found by many, many thousands of other make processes in this long compilation. Why it should disappear for one, and only one, call among the galaxy of other calls, I don't know.

I also received two "fork: invalid argument" notices. Again, restarting the make process recovered from this. This might be #2469 again.

So, this is much, much more reliable. But the "command not found" is a mystery.

**Comment by bitcrazed** [comment]
@RinatMullayanov - we'll let you know once the fix bubbles up into the servicing release mechanisms.

**Comment by bitcrazed** [comment]
@Warblefly - wow - 36 hour compile??? ;) Have asked the team to take a look at #2469 which is similar to the issue you describe - please follow/contribute there.

**Comment by TomasHubelbauer** [comment]
I wish I didn't have to connect Microsoft Account with my installation of Windows in order to participate in Windows Insider. I could really use this fix ASAP, not whenever this is scheduled to go out to non-Insiders. :-(

**Comment by Stanzilla** [comment]
You can download an ISO

**Comment by hoichi** [comment]
Or you can download a Linux ISO while you’re at it. I know I did, and the console works significanlty faster, by the way. Still boot to Windows, but not at work. It’s a shame; miss it a lot.

Seriously folks, I don’t want to sound bitter, but people got other things to do than downloading ISOs or installing insider builds. I know they’re pretty stable, but still.

I know the WSL team probably has very little influence on the release cycle, but I think those who have should reconsider the criticality of the fix. You risk losing people, people.

**Comment by Sendoushi** [comment]
I'm having the same issue when trying to use `yarn`. It is blocking my environment. I might need to change OSes or use docker just because of this :s

**Comment by Stanzilla** [comment]
@bitcrazed @SvenGroot any reason you guys don't use the Release Preview ring for stuff like that? Seems ideal to me.

**Comment by bitcrazed** [comment]
@TomasHubelbauer The reason we ask you to primarily get your distros from the store is that this allows distro vendors to patch and service their releases, ensuring that when you install a distro, you're always installing the latest supported version. This requires that you sign-in with your MSA, so that we know to push updates to your machines when they're published.

@hoichi The distros that you download and install manually are identical to the distros you install via the Store so there'll be no perf differential between the two.

@Stanzilla Let's just say that the Windows servicing release mechanism is complex, for good reasons ;)

**Comment by SvenGroot** [comment]
The fix for the Fall Creator's Update should be coming very soon. We can't comment on the timeline, because the patch could be delayed for any number of reasons beyond our control. In the meantime, the workaround posted above should prevent any issues.

We appreciate your patience and understanding.

**Comment by nutcasev15** [comment]
@SvenGroot 
I'm on build 16299.64 . Is this issue related to #2464 ? Will that be fixed in the incoming backport patch?

**Comment by therealkenc** [comment]
@nutcasev15  - #2464 went off the reserve because it started as what looked like as a LxFs problem and then someone posted a DrvFs problem in the same thread. The original LxFs problem is still in an indeterminate "needs repro" state. I compile huge GCC builds on LxFs regularly without `EINVAL` issues so there is no way to tell what is going on over there or whether it is "related". The 17035 release notes for **this** issue say DrvFs, and I'd take that at face value.

**Comment by SvenGroot** [comment]
**Good news, everyone!**

The November 30th, 2017 Cumulative Update (KB4051963) for the Fall Creator's Update contains the fix for this issue. Please install this update, and let us know if you encounter any further issues.

To verify if you have installed the update, your OS build number should be 16299.98 or later.

**Comment by sisou** [comment]
The update requires a computer reboot, but afterwards it _works for me_.

**Comment by Warblefly** [comment]
A quick summary of #2712: the files that intermittently go missing are simply "not found". This isn't an EINVAL error. Not sure if it's exactly related to #2448.

**Comment by codeart1st** [comment]
Works for me too, no issues anymore

**Comment by mindhaq** [comment]
after applying the update, the error is no more.

but - was this always so slow? `create-react-app test` takes over 62 seconds in bash compared to 16 seconds in cmd on a decent i7 4.2Ghz workstation.

**Comment by rhyek** [comment]
Yeah, I hadn't used WSL for months. Trying it now and adding packages using `yarn` is incredibly slow for my project. I usually work on it on my dual booted Ubuntu install and it is definitely quicker there.

**Comment by bitcrazed** [comment]
@rhyek @mindhaq Disk IO perf is a well known WSL issue that Sven is working with the storage team to address in future WSL releases.

However, while WSL disk perf is slower right now, it should work correctly. Can you confirm?

**Comment by rhyek** [comment]
@bitcrazed Yes, everything is functional.

**Comment by bitcrazed** [comment]
Great thanks.

Closing for now since we have a few confirmations that the issue appears to be resolved.

**Comment by mindhaq** [comment]
@bitcrazed Apart from slow IO it is working fine now, thank you!

It's good to hear that disk perf is being addressed. Is there an Issue to follow on?

**Comment by bitcrazed** [comment]
Thanks for letting us know. 

The larger disk perf issue is primarily being tracked in issue #873

**Comment by shjyh** [comment]
I upgrade my OS to newest insider preview(17025.10000)，still have this problem

**Comment by therealkenc** [comment]
> newest insider preview(17025.10000)

[this](https://blogs.windows.com/windowsexperience/2017/11/22/announcing-windows-10-insider-preview-build-17046-pc/#xzjq55j7cpGGhQOB.97)

**Comment by Nadav-Ruskin** [comment]
I am using Windows version 1803, build 17134.112 and still encountering this bug.

**Comment by muyu66** [comment]
@Nadav-Ruskin  NPM 6.0 is recommended, It is no worse than yarn now

**Comment by mrmckeb** [comment]
That's not really a fair answer, @muyu66. First, it's not true that npm@6 matches Yarn in terms of functionality. More importantly, WSL is not meant to be limited to a subset of tools, and you can't expect development teams to drop Yarn because none is "recommended".

**Comment by Nadav-Ruskin** [comment]
Sorry if I haven't made myself clear: The error I'm experiencing is more related to #2606, that is, I can't build (my own projects) using `make -j<anything larger than 1>`. I've posted about it here because this thread contains the workaround and claims that the issue has been fixed.
I'll make a new issue to avoid further confusion.

**Comment by onomatopellan** [comment]
@tomitrescak what build? (Win+R -> type `winver`)
Your OS build number should be 16299.98 or later to get this fix.

**Comment by tomitrescak** [comment]
Ah, ok, I got 16299.64, installing update now.
